pkg/controller/strategy/comparisons.go

```diff
@@ -1,11 +1,23 @@
 package strategy
 
-func contenderCapacityComparison(achieved uint, desired uint) bool {
-	return achieved >= desired
+func contenderCapacityComparison(achievedPercentage, desiredPercentage, totalNumberOfPods uint) bool {
+	// Consider that contender capacity has achieved its desired state if
+	// the end game is one pod and we already have it, as long as desired
+	// is greater than zero.
+	if totalNumberOfPods == 1 && achievedPercentage == 100 && desiredPercentage > 0 {
+		return true
+	}
+	return achievedPercentage >= desiredPercentage
 }
 
-func incumbentCapacityComparison(achieved uint, desired uint) bool {
-	return achieved <= desired
+func incumbentCapacityComparison(achievedPercentage, desiredPercentage, totalNumberOfPods uint) bool {
+	// Consider that the incumbent capacity has achieved its desired state if
+	// the end game is one pod and we already have it, as long as the desired
+	// is smaller than 100.
+	if totalNumberOfPods == 1 && achievedPercentage == 100 && desiredPercentage < 100 {
```

Comment:
I don't totally understand the  `desired < 100` check; my first impression is that it is a bug, since it might say its OK to proceed when we should actually be scaling down to 0. 

Reasoning: if `desiredPercentage` is 0 on a 1 pod deployment, and we (for whatever reason) haven't made any progress turning off that pod, I think this will say that everything is correct, when really it should say that we're not there yet. I think the correct test might be: `totalPods==1 && achievedPercent == 100 && desiredPercentage > 0` for the incumbent as well.